spec/market_spec.rb

```diff
@@ -0,0 +1,117 @@
+require './lib/item'
+require './lib/vendor' 
+require './lib/market'
+
+RSpec.describe Market do
+    before(:each) do
+        @vendor1 = Vendor.new("Rocky Mountain Fresh")
+        @vendor2 = Vendor.new("Ba-Nom-a-Nom")
+        @vendor3 = Vendor.new("Palisade Peach Shack")
+        @vendor4 = Vendor.new("Choco Monster")
+        @item1 = Item.new({name: 'Peach', price: "$0.75"})
+        @item2 = Item.new({name: 'Tomato', price: "$0.50"})
+        @item3 = Item.new({name: 'Peach-Raspberry Nice Cream', price: "$5.30"})
+        @item4 = Item.new({name: 'Banana Nice Cream', price: "$4.25"})
+        @market = Market.new("South Pearl Street Farmers Market")
+    end
+
+    describe '#initialize' do
+        it "exists" do
+            expect(@market).to be_instance_of(Market)
+        end
+
+        it 'has a name' do
+            expect(@market.name).to eq("South Pearl Street Farmers Market")
+        end
+
+        it 'has a date' do
+            expect(@market.date).to eq("2024/09/16")
+        end
+
+        it 'starts with an empty array of vendors' do
+            expect(@market.vendors).to eq([])
+        end
+    end
```

Comment:
Functionality from the initialize method can be consolidated into a single test to save time and lines of code.